src/main/kotlin/ecommerce/client/StripeClient.kt

```diff
@@ -0,0 +1,57 @@
+package ecommerce.client
+
+import ecommerce.config.StripeProperties
+import ecommerce.dto.PaymentIntentResponse
+import ecommerce.dto.PaymentRequest
+import ecommerce.dto.mapStringToStripeErrorResponse
+import ecommerce.dto.toPaymentIntentResponse
+import ecommerce.exception.BadRequestException
+import ecommerce.exception.ExternalServiceException
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestClient
+import org.springframework.web.client.RestClientResponseException
+
+@Component
+class StripeClient(
+    private val stripeProperties: StripeProperties,
+    private val restClient: RestClient,
+) {
+    fun createCheckoutSession(req: PaymentRequest): PaymentIntentResponse {
+        val body =
+            listOf(
+                "amount=${req.amount}",
+                "currency=${req.currency}",
+                "payment_method=${req.paymentMethod}",
+                "confirm=true",
+                "automatic_payment_methods[enabled]=true",
+                "automatic_payment_methods[allow_redirects]=never",
+            ).joinToString("&")
+
+        return try {
+            val response =
+                restClient.post()
+                    .uri("https://api.stripe.com/v1/payment_intents")
```

Comment:
I recommend managing 3rd party api URL in a different file such as in properties file (`application.properties` could be an option), and inject the value in application logic.

This is because by separating the URL from application logic, we can easily handle the changes when the URL changes, and we can also mock the API when testing.